app/src/main/java/org/dhis2/usescases/datasets/datasetInitial/DataSetInitialActivity.java

```diff
@@ -180,6 +181,21 @@ public void showCatComboSelector(String catOptionUid, List<CategoryOption> data)
                         checkActionVisivbility();
                     })
                     .show(this, selectedView);
+        }else{
+            new CategoryDialog(
+                    CategoryDialog.Type.CATEGORY_OPTIONS,
+                    catOptionUid,
+                    true,
+                    selectedPeriod,
+                    selectedOption -> {
+                        CategoryOption categoryOption = presenter.getCatOption(selectedOption);
+                        selectedCatOptions.put(catOptionUid, categoryOption);
+                        ((TextInputEditText) selectedView).setText(categoryOption != null ? categoryOption.displayName() : null);
+                        checkActionVisivbility();
+                        return null;
```

Comment:
why null?

Reply:
its a kotlin function ()->Unit, which in java requires returning a null.

Reply:
I think you can return Unit.INSTANCE?